### PR TITLE
fix(NotificationList): a11y improvements

### PR DESCRIPTION
**Pull request by saghdaey**
### Overview

Adds aria-hidden to the decorative image (that appears when the user has no notifications)
Wraps NotificationList in an aria-live polite
Improves aria-labels on some notification elements 

monolith PR: https://github.com/codecademy-engineering/Codecademy/pull/27102

EGG-917:
from Test IO bug ticket: When user presses the *down arrow* the expected focussing of the *You're all caught up! Go forth and learn new things* fails and screen reader announces the *Notification drop down* as *Blank*. 

^^~I've managed to fix the screen reader announcing "blank" and instead announcing that the user is caught up, but the focus is still remaining on where the clear button used to be.. and i havent been able to fix that yet~

fixed it!
### PR Checklist

- [ ] Related to designs:
- [X] Related to JIRA ticket: [EGG-917,](https://codecademy.atlassian.net/browse/EGG-917) [EGG-918](https://codecademy.atlassian.net/browse/EGG-918)
- [x] I have run this code to verify it works
- [ ] This PR includes unit tests for the code change

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.5.78-alpha.6d11c4.0<br/>@codecademy/gamut-labs@19.26.7-alpha.6d11c4.0<br/>@codecademy/gamut@37.4.2-alpha.6d11c4.0<br/>@codecademy/monaco@0.4.5-alpha.6d11c4.0<br/>@codecademy/styleguide@44.5.4-alpha.6d11c4.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://61420696ce501e5d49a5ab84--gamut-preview.netlify.app>https://61420696ce501e5d49a5ab84--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/61420696ce501e5d49a5ab84>Deploy Logs</a></p>

File: packages/gamut-labs/src/Notifications/NotificationsContents.tsx [hunk 1]
```diff
@@ -32,7 +32,11 @@ export const NotificationsContents: React.FC<NotificationsContentsProps> = (
 
   return (
     <Background
-      aria-label={`My ${notifications.length} notifications`}
+      aria-label={
+        notifications.length
+          ? `My ${notifications.length} notifications`
+          : `My Notifications`
+      }
```
Comment (saghdaey) [inline packages/gamut-labs/src/Notifications/NotificationsContents.tsx:39]: so it doesn't read out "My 0 Notifications"

File: packages/gamut/src/NotificationList/index.tsx [hunk 2]
```diff
@@ -26,22 +26,26 @@ export const NotificationList: React.FC<NotificationListProps> = ({
...
-        <NotificationItem
-          key={notification.id}
-          notification={notification}
-          handleClick={() => onNotificationClick?.(notification)}
-          handleDismiss={() => onDismiss?.(notification)}
-        />
-      ))}
-    </UnstyledUnorderedList>
+  return (
+    <div aria-live="polite">
```
Comment (saghdaey) [inline packages/gamut/src/NotificationList/index.tsx:30]: this way if a user had 3 notifications and clicks clear all, the empty notifications content will be read

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.5.79-alpha.82afa5.0<br/>@codecademy/gamut-labs@19.26.8-alpha.82afa5.0<br/>@codecademy/gamut@37.5.1-alpha.82afa5.0<br/>@codecademy/monaco@0.4.6-alpha.82afa5.0<br/>@codecademy/styleguide@44.6.1-alpha.82afa5.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://61422a1342e8a4157e977822--gamut-preview.netlify.app>https://61422a1342e8a4157e977822--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/61422a1342e8a4157e977822>Deploy Logs</a></p>

**Comment by saghdaey** [comment]
I'm still seeing an issue where focus is remaining on the Clear Button element (which actually gets removed after you press it, since it clears all the users notifications)

Screen Shot 2021-09-15 at 2 12 01 PM

**Review by JoshuaKGoldberg — APPROVED** [review]
💖

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.5.79-alpha.c75bea.0<br/>@codecademy/gamut-labs@19.26.8-alpha.c75bea.0<br/>@codecademy/gamut@37.5.1-alpha.c75bea.0<br/>@codecademy/monaco@0.4.6-alpha.c75bea.0<br/>@codecademy/styleguide@44.6.1-alpha.c75bea.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6142591fa54b170a0b57518a--gamut-preview.netlify.app>https://6142591fa54b170a0b57518a--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6142591fa54b170a0b57518a>Deploy Logs</a></p>

**Comment by saghdaey** [comment]
fixed the focus thing! (i think)
now this is in focus after you clear all of your. notifications 

Screen Shot 2021-09-15 at 4 39 11 PM

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.5.80-alpha.545c58.0<br/>@codecademy/gamut-labs@20.0.1-alpha.545c58.0<br/>@codecademy/gamut@37.5.1-alpha.545c58.0<br/>@codecademy/monaco@0.4.6-alpha.545c58.0<br/>@codecademy/styleguide@44.6.2-alpha.545c58.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://61425abd0bbc8400aa1a8705--gamut-preview.netlify.app>https://61425abd0bbc8400aa1a8705--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/61425abd0bbc8400aa1a8705>Deploy Logs</a></p>

**Review by JoshuaKGoldberg — APPROVED** [review]
Yup!

**Review by katiezutter — APPROVED** [review]
Yay! LGTM

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.5.81-alpha.4914ba.0<br/>@codecademy/gamut-labs@20.0.2-alpha.4914ba.0<br/>@codecademy/gamut@37.5.2-alpha.4914ba.0<br/>@codecademy/monaco@0.4.7-alpha.4914ba.0<br/>@codecademy/styleguide@44.6.3-alpha.4914ba.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6143728e36942030ffde9d66--gamut-preview.netlify.app>https://6143728e36942030ffde9d66--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6143728e36942030ffde9d66>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.5.81-alpha.78d2a4.0<br/>@codecademy/gamut-labs@20.0.2-alpha.78d2a4.0<br/>@codecademy/gamut@37.5.2-alpha.78d2a4.0<br/>@codecademy/monaco@0.4.7-alpha.78d2a4.0<br/>@codecademy/styleguide@44.6.4-alpha.78d2a4.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6143799361068e0099e64712--gamut-preview.netlify.app>https://6143799361068e0099e64712--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6143799361068e0099e64712>Deploy Logs</a></p>

**Pull request merged**